rectools/models/sasrec.py

```diff
@@ -0,0 +1,723 @@
+import logging
+import typing as tp
+import warnings
+from copy import deepcopy
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+import torch
+import tqdm
+from torch import nn
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset as TorchDataset
+
+from rectools import Columns, ExternalIds
+from rectools.dataset import Dataset, Interactions
+from rectools.dataset.identifiers import IdMap
+from rectools.models.base import ErrorBehaviour, InternalRecoTriplet, ModelBase
+from rectools.models.rank import Distance, ImplicitRanker
+from rectools.types import InternalIdsArray
+
+PADDING_VALUE = "PAD"
+
+logger = logging.getLogger(__name__)
```

Comment:
It's nice to have logging, but we need a proper setup.  
Now it's used only in one place and quite useless. Let's remove it for now

Reply:
Ok, then please just add TODO

Reply:
fixed